CHANGELOG.md

```diff
@@ -7,15 +7,39 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Provide `ServerTestAppExt` extension trait for exchanging messaging between apps in tests.
+
 ### Changed
 
-- Rename `ClientCache` into `ConnectedClients`.
-- Rename `ClientState` into `ConnectedClient`.
+- Abstract out all I/O and `renet` dependency. We will continue to provide first-party integration with renet via `bevy_replion_renet`. But users can write their integration with other messaging libraries. So now users need to additionally add messaging-related plugin. In case of `bevy_replion_renet` it's `RepliconRenetPlugins`.
+- Replace usage of `RenetServer` and `RenetClient` with our `RepliconServer` and `RepliconClient` respectively. Use types from `renet` (or other library) only when you need to connect or write some library-specific logic. In other cases prefer using the newly provided types to make your code messaging-library independent. Unlike the old types from renet, these resources are always present in the world. So instead of using `resource_(exists/added/removed)` for network-related conditions, use special conditions provided in `common_conditions` module.
```

Comment:
> I would not rename ClientId to PeerId

It's not a rename, `ClientId` comes from `renet`, I can't use it. So I had to come up with my own struct.

I decided to use the word `Peer` because it can be applied for client and server (ID == 0). Messaging libraries unlikely use this name, this is why `Replicon` is unnecessary in this case. The word `peer` in server-authoritative context usually used by more high level libraries. For example, Godot uses this convention. [Here](https://docs.godotengine.org/en/stable/classes/class_multiplayerapi.html#signals) is how server events look.

Clients still have `ClientId`, you just need to obtain it from `renet`.

Reply:
> Users of the library should not need to use renet's ClientId. 

Users pass client ID to renet's `ClientAuthentication`, but it's not a `ClientId`, it's just integer...
So I guess we could do it.

Godot targets server-authoritative, but I see how this can be confused. I will rename it into `ClientId`.

Reply:
Done! Reverted all related changes.